File: wk04/3-wed/max_num_vowel_count.js
// Homework 17th of April 2019

// 1. Max of two numbers:
// Define a function maxOfTwoNumbers that takes two numbers as arguments and returns the largest of them.
// Use the if-then-else construct available in JavaScript. Do some Googling to figure this out if you
// forget how conditionals work.

// 2. Vowel Count:
// Return the number of occurrences of vowels in a word. Vowels are a, e, i, o, u, and y

// Function signature (only look at this after you've thought about what it should look like)


var maxOfTwoNumbers = function(num1, num2) {
  if (num1 > num2) {
    return num1;
  } else {
    return num2;
  }
};
maxOfTwoNumbers(23, 5);

var signature = function(word) {
  var vowels = word.match(/[aeiouy]/gi); // g - (g)lobal, i - case (i)nsensitive

  if (vowels === null) {
    return 0;
  } else {
    return vowels.length;
  }
};

signature("Australia");

File: wk04/3-wed/max_num_vowel_count.rb
#Homework 17th April 2019

# Question One
def greater (num1, num2)
  if num1 > num2
    puts "#{num1} is greater than #{num2}"
  else 
    puts "#{num2} is greater than #{num1}"
  end
end
greater(5,8)


# Question 2

def signature (word) 
  return word.scan(/[aeiouy]/i).length
end
puts signature("xyz")
